File: app/src/main/java/com/wilinz/yuetingmusic/util/LogUtil.java
package com.wilinz.yuetingmusic.util;

import android.util.Log;

/**
 * Created by root on 15-11-2.
 */
public class LogUtil {

    public static final int VERBOSE = 1;
    public static final int DEBUG = 2;
    public static final int INFO = 3;
    public static final int WARN = 4;
    public static final int ERROR = 5;
    public static final int NOTHING = 6;

    public static final int LEVEL = VERBOSE;

    public static void v(String tag, Object msg) {
        if (LEVEL <= VERBOSE) {
            Log.v(tag, msg.toString());
        }
    }

    public static void d(String tag, Object msg) {
        if (LEVEL <= DEBUG) {
            Log.d(tag, msg.toString());
        }
    }

    public static void i(String tag, Object msg) {
        if (LEVEL <= INFO) {
            Log.i(tag, msg.toString());
        }
    }

    public static void w(String tag, Object msg) {
        if (LEVEL <= WARN) {
            Log.w(tag, msg.toString());
        }
    }

    public static void e(String tag, String msg) {
        if (LEVEL <= ERROR) {
            Log.e(tag, msg.toString());
        }
    }
}

File: app/src/main/java/com/wilinz/yuetingmusic/util/LogUtil.kt
package com.wilinz.yuetingmusic.util

import android.util.Log

/**
 * Created by root on 15-11-2.
 */
object LogUtil {
    const val VERBOSE = 1
    const val DEBUG = 2
    const val INFO = 3
    const val WARN = 4
    const val ERROR = 5
    const val NOTHING = 6
    const val LEVEL = VERBOSE
    fun v(tag: String?, msg: Any) {
        if (LEVEL <= VERBOSE) {
            Log.v(tag, msg.toString())
        }
    }

    @JvmStatic
    fun d(tag: String?, msg: Any) {
        if (LEVEL <= DEBUG) {
            Log.d(tag, msg.toString())
        }
    }

    fun i(tag: String?, msg: Any) {
        if (LEVEL <= INFO) {
            Log.i(tag, msg.toString())
        }
    }

    fun w(tag: String?, msg: Any) {
        if (LEVEL <= WARN) {
            Log.w(tag, msg.toString())
        }
    }

    fun e(tag: String?, msg: String) {
        if (LEVEL <= ERROR) {
            Log.e(tag, msg)
        }
    }
}